pyresample/test/test_spherical.py

```diff
@@ -308,6 +325,41 @@ def test_angle(self):
         arc2 = Arc(SCoordinate(2, 0), SCoordinate(3, 0))
         self.assertRaises(ValueError, arc1.angle, arc2)
 
+    def test_arc_crossing_antimeridian(self):
+        import copy
+
+        # Tests disjoint arcs
+        arc1 = Arc(SCoordinate(*np.deg2rad((143.76, 0))),
+                   SCoordinate(*np.deg2rad((143.95, 7.33)))
+                   )
+        arc2 = Arc(SCoordinate(*np.deg2rad((170.34, 71.36))),
+                   SCoordinate(*np.deg2rad((-171.03, 76.75)))
+                   )
+        arc1_orig = copy.deepcopy(arc1)
+        arc2_orig = copy.deepcopy(arc2)
+        point = arc1.intersection(arc2)
+        # Assert original arcs are unaffected
+        assert arc1_orig.end.lon == arc1.end.lon
+        assert arc2_orig.end.lon == arc2.end.lon
+        # Assert disjoint arcs returns None
+        assert isinstance(point, type(None))
+
+        # Test intersecting arcs across the antimeridian
```

Comment:
Can you split this second half into another test method? Or we could get fancy and I could show you how to do pytest parametrize stuff, but we'd have to put this test in another (non-unittest.TestCase) class.

Reply:
Now I split into two functions. But would be nice if we can meet once and you explain to me briefly the best practices so that I can use them for the new spherical tests ...

Reply:
Yes, I could meet if needed. I'm not sure I have too much to teach, but if you're OK with asynchronous chatting we could talk about this in the pytroll slack's `#code` channel.

As for this suggestion, it may not be obvious why I asked to do this compared to other tests in pyresample. A lot of the existing tests in pyresample are following bad practice. Additionally a lot of the tests are written with the standard library's `unittest` which is not as flexible compared to pytest which we've been using more and more throughout the pytroll libraries. In general, we want to keep clean and short test methods so they are easy to understand and easy to modify if needed. This should result in less duplicated code if the tests all depend on the same helper functions (for example, to create fake data). It also helps in seeing at a glance what is failing and why if the test has a single purpose. For example, if I run the pyresample tests and your original function fails, I don't know (initially) if the first chunk failed or the second. And even if I did I have to read the source code and understand the entire test function to understand how the various sections differ. The name of the test function/method is usually shown when there is a failure so if I see that and it is descriptive I can usually get a basic idea of what is going on and why the tests are failing.

Reply:
OK with asynchronous chatting we could talk about this in the pytroll slack's #code channel. 
Just create/add me to that channel ;)

Reply:
I think you can join whatever channels you want on the slack :wink: